apps/server/src/controllers/category.controller.ts

```diff
@@ -0,0 +1,101 @@
+import { Request, Response, NextFunction } from 'express';
+import {
+  CreateCategoryInput,
+  UpdateCategoryInput,
+} from '../schemas/category.schema';
+import { CategoryService } from '../services/category.service';
+
+export class CategoryController {
+  private service: CategoryService;
+
+  constructor(_service: CategoryService) {
+    this.service = _service;
+  }
+
+  async getCategories(req: Request, res: Response, next: NextFunction) {
+    try {
+      const limit = Number(req.query['limit']);
+      const offset = Number(req.query['offset']);
+      const categories = await this.service.getCategories(limit, offset);
+      return res.status(200).json({
+        status: 'success',
+        data: { categories },
+      });
+    } catch (error) {
+      next(error);
+    }
+  }
+
+  async getCategoryById(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { id } = req.params;
+      const category = await this.service.getCategoryById(id);
+      if (!category) {
```

Comment:
move this logic inside service and throw NotFoundError() exception:

if (!category) {
   throw new NotFoundError('Category not found');
}